packages/stencil-library/src/components/alert/alert.e2e.ts

```diff
@@ -1,10 +1,119 @@
+    expect(slotContainer).toHaveClass('usa-alert__text');
+    const slottedContent = await slotContainer.find('span');
+    expect(slottedContent).toEqualText('Body alert text');
+  });
+
+  it('applies no-icon class when no-icon is true', async () => {
+    const { alertDiv } = await renderAndGetAlertDiv(`
+      <njwds-alert no-icon>
+        <span>Body alert text</span>
+      </njwds-alert>
+    `);
+    expect(alertDiv).toHaveClass('usa-alert--no-icon');
+  });
+
+  describe('type', () => {
+    it('renders without special styling by default', async () => {
+      const { alertDiv } = await renderAndGetAlertDiv(`
+        <njwds-alert>
+          <span>Body alert text</span>
+        </njwds-alert>
+      `);
+      expect(alertDiv.className).toBe('usa-alert');
+    });
+
+    it.each(['info', 'warning', 'error', 'emergency'])('renders type %s when passed', async type => {
+      const { alertDiv } = await renderAndGetAlertDiv(`
+        <njwds-alert type=${type}>
+          <span>Body alert text</span>
+        </njwds-alert>
+      `);
+      expect(alertDiv).toHaveClass(`usa-alert--${type}`);
+    });
+  });
+
+  describe('role', () => {
+    it.each(['error', 'emergency'])('sets role alert for %s', async type => {
+      const { alertDiv } = await renderAndGetAlertDiv(`
+        <njwds-alert type=${type}>
+          <span>Body alert text</span>
+        </njwds-alert>
+      `);
+      expect(alertDiv.getAttribute('role')).toEqual('alert');
+    });
+
+    it.each(['info', 'warning'])('does not set role for %s', async type => {
+      const { alertDiv } = await renderAndGetAlertDiv(`
+        <njwds-alert type=${type}>
+          <span>Body alert text</span>
+        </njwds-alert>
+      `);
+      expect(alertDiv.getAttribute('role')).toBeNull();
+    });
+  });
+
+  describe('header slot', () => {
+    it("renders without a header when the header slot isn't provided", async () => {
+      const { alertDiv } = await renderAndGetAlertDiv(`
+        <njwds-alert>
+          <span>Body alert text</span>
+        </njwds-alert>
+      `);
+      const header = await alertDiv.find('[slot="header"]');
+      expect(header).toBeNull();
+    });
+
+    it('renders a header when the header slot is provided', async () => {
+      const { alertDiv } = await renderAndGetAlertDiv(`
+        <njwds-alert>
+          <h3 slot="header">Heading Text</h3>
+          <span>Body alert text</span>
+        </njwds-alert>
+      `);
+      const header = await alertDiv.find('[slot="header"]');
+      expect(header).toHaveClass('usa-alert__heading');
+      expect(header).toEqualText('Heading Text');
+    });
+  });
+
+  describe('is-slim', () => {
+    it('adds slim when slim is true', async () => {
+      const { alertDiv } = await renderAndGetAlertDiv(`
+        <njwds-alert slim>
+          <span>Body alert text</span>
+        </njwds-alert>
+      `);
+      expect(alertDiv).toHaveClass(`usa-alert--slim`);
```

Comment:
[pebble] same comment as no-icon above re testing for no slim class when slim is false or not provided.